src/execution_plan/ops/op_conditional_traverse.c

```diff
@@ -218,7 +218,11 @@ static Record CondTraverseConsume(OpBase *opBase) {
 
 static OpResult CondTraverseReset(OpBase *ctx) {
 	CondTraverse *op = (CondTraverse *)ctx;
-	if(op->r) OpBase_DeleteRecord(op->r);
+
+	op->r = NULL;
```

Comment:
Leak? why aren't we freeing `OpBase_DeleteRecord(op->r);` ?

Reply:
`op->r` is not an independent pointer; it is also one of the entries in `op->records` and as such will be freed in the loop immediately following here.

I will add an explanatory comment here.